pkg/util/labels.go

```diff
@@ -179,9 +179,14 @@ func addDeploymentConfigNestedLabels(obj *deployapi.DeploymentConfig, labels lab
 }
 
 func addDeploymentConfigNestedAnnotations(obj *deployapi.DeploymentConfig, annotations map[string]string) error {
+	if obj.Spec.Template == nil {
+		return fmt.Errorf("DeploymentConfig \"%s\" is invalid:\n\n* spec.Template: Required value", obj.Name)
+	}
```

Comment:
i'm not convinced this is the right fix, because we really shouldn't be doing our own validation of the resource.  We should just be more defensive in our own code such that when we go to set the label, if things are nil, we do not set the label (and perhaps print a warning about it).

The user will ultimately get the validation error from the api server when we go to create this object, and I'd prefer to keep the validation there, so it's in one place.  Otherwise we're on a slippery slope to start doing other validation checks on the resources from within new-app, and possibly getting out of sync on what is considered valid by new-app vs valid according to the api server.